finagle-http/src/main/scala/com/twitter/finagle/http/codec/ResponseConformanceFilter.scala

```diff
@@ -36,13 +39,57 @@ private[codec] object ResponseConformanceFilter extends SimpleFilter[Request, Re
   private[this] def validate(req: Request, rep: Response): Unit = {
     if (req.method == Method.Head) {
       handleHeadResponse(req, rep)
+    } else if (mustNotIncludeMessageBody(rep.status)) {
+      handleNoMessageResponse(rep)
     } else if (rep.isChunked) {
       handleChunkedResponse(rep)
     } else {
       handleFullyBufferedResponse(rep)
     }
   }
 
+  /**
+   * 1. To conform to the RFC, a message body is removed if a status code is either 1xx, 204 or 304.
+   *
+   * RFC7230 section-3.3: (https://tools.ietf.org/html/rfc7230#section-3.3)
+   * "All 1xx (Informational), 204 (No Content), and 304 (Not Modified) responses do not include a message body."
+   *
+   * 2. Additionally, a Content-Length header field is dropped for 1xx and 204 responses as described in RFC7230
+   * section-3.3.2. It, however, is allowed to send a Content-Length header field in a 304 response. To follow
+   * the section, we don't remove the header field from a 304 response but its value is not checked nor corrected.
+   *
+   * RFC7230 section-3.3.2: (https://tools.ietf.org/html/rfc7230#section-3.3.2)
+   * "A server MUST NOT send a Content-Length header field in any response with a status code of 1xx (Informational)
+   * or 204 (No Content)."
+   *
+   * "A server MAY send a Content-Length header field in a 304 (Not Modified) response to a conditional GET request
+   * (Section 4.1 of [RFC7232]); a server MUST NOT send Content-Length in such a response unless its field-value equals
+   * the decimal number of octets that would have been sent in the payload body of a 200 (OK) response to the same
+   * request."
+   */
+  private[this] def handleNoMessageResponse(rep: Response): Unit = {
```

Comment:
Do you mean we need a plan to conform to [RFC7230#section-3.3.1](https://tools.ietf.org/html/rfc7230#section-3.3.1) probably?
It says:
```
   A server MUST NOT send a Transfer-Encoding header field in any
   response with a status code of 1xx (Informational) or 204 (No
   Content).  A server MUST NOT send a Transfer-Encoding header field in
   any 2xx (Successful) response to a CONNECT request (Section 4.3.6 of
   [RFC7231]).
```

Do you think it's better for us to handle the issue in this PR or another PR?

Well, my first thought to solve the problem is like this:
```
private[this] def handleNoMessageResponse(rep: Response): Unit = {
  // remove message body if necessary

  // remove Content-Length and Transfer-Encoding if necessary
  if (rep.status != 304 && mustNotIncludeMessageBody(rep.status)) {
  	if (rep.contentLength.isDefined) {
      rep.headerMap.remove(Fields.ContentLength)
      log.error(...)
  	}
  	if (rep.isChunked) {
      rep.headerMap.remove(Fields.TransferEncoding)
      log.error(...)
  	}
  }
}
```